src/test/java/bio/terra/app/controller/DatasetConnectedTest.java

```diff
@@ -51,36 +59,96 @@
 
     @MockBean private IamService samService;
 
-    @Test
-    public void testCreateOmopDataset() throws Exception {
+    private BillingProfile billingProfile;
+
+    @Before
+    public void setup() throws Exception {
         connectedOperations.stubOutSamCalls(samService);
-        String accountId = googleResourceConfiguration.getCoreBillingAccount();
+        billingProfile = ProfileFixtures.billingProfileForAccount(googleResourceConfiguration.getCoreBillingAccount());
+        UUID profileId = profileDao.createBillingProfile(billingProfile);
+        billingProfile.id(profileId);
+    }
 
+    @After
+    public void tearDown() throws Exception {
+        profileDao.deleteBillingProfileById(billingProfile.getId());
+    }
+
+    @Test
+    public void testCreateOmopDataset() throws Exception {
         DatasetRequestModel datasetRequest = jsonLoader.loadObject("it-dataset-omop.json", DatasetRequestModel.class);
-        UUID profileId = profileDao.createBillingProfile(ProfileFixtures.billingProfileForAccount(accountId));
         datasetRequest
             .name(Names.randomizeName(datasetRequest.getName()))
-            .defaultProfileId(profileId.toString());
+            .defaultProfileId(billingProfile.getId().toString());
 
+        MockHttpServletResponse response = createDataset(datasetRequest, status().isCreated());
+        assertThat("create omop dataset successfully", response.getStatus(), equalTo(HttpStatus.CREATED.value()));
+        DatasetSummaryModel datasetSummaryModel =
+            objectMapper.readValue(response.getContentAsString(), DatasetSummaryModel.class);
+
+        response = deleteDataset(datasetSummaryModel.getId());
+        checkDeleteSuccessful(response);
+    }
+
+    @Test
+    public void testDuplicateName() throws Exception {
+        // create a dataset and check that it succeeds
+        DatasetRequestModel datasetRequest = jsonLoader
+            .loadObject("snapshot-test-dataset.json", DatasetRequestModel.class);
+        datasetRequest
+            .name(Names.randomizeName(datasetRequest.getName()))
+            .defaultProfileId(billingProfile.getId().toString());
+
+        MockHttpServletResponse response = createDataset(datasetRequest, status().isCreated());
+        assertThat("created test dataset successfully", response.getStatus(), equalTo(HttpStatus.CREATED.value()));
+        DatasetSummaryModel datasetSummaryModel =
+            objectMapper.readValue(response.getContentAsString(), DatasetSummaryModel.class);
+
+        // try to create the same dataset again and check that it fails
+        response = createDataset(datasetRequest, status().is5xxServerError());
```

Comment:
My intuition is that this should result in a `4xx` bad request error instead of an internal server error. What do you think?

Reply:
I agree. Will make the change and update.